src/bend.ts

```diff
@@ -280,6 +283,6 @@ export class Bend extends Modifier {
     } else if (lastBend.type === Bend.DOWN) {
       renderArrowHead(lastBend.x + lastDrawnWidth, start.y, -1);
     }
-    this.restoreStyle();
+    ctx.restore();
```

Comment:
Is this ctx.restore() paired with the ctx.save() from line 174? It's a big method so I wanted to make sure the save/restore calls are all correctly matched.

Reply:
I don't think it's such a big deal, but if we cared about making sure things were correctly matched, we could make ctx.save() return an object like:

```
const savedContext = ctx.save();
...
savedContext.restore();
```

where ctx.save returns:
```
 {
  restore:  () => { 
    if (this._ctx) { 
      this._ctx.restore(); 
      this._ctx = null; 
    }
  },
  _ctx: this,
}
```

Then we don't have to worry about a context being restored twice, or from a non-existent save, and failing to restore context will give a typescript warning that `savedContext` was defined but never used.

Reply:
@ronyeh thanks a lot for your review! yes, this ctx.restore() is paired with the ctx.save() from line 174
@mscuthbert the canvas context does not return a value, see https://www.w3schools.com/jsref/canvas_save.asp. I believe that the svg context should not be have differently.

My proposal in #223 reduces a lot the number of calls to ´ctx.save()´ I will raise a PR.